caravel/views.py

```diff
@@ -1964,6 +1964,38 @@ def cached_key(self, key):
         return "nope"
 
     @has_access_api
+    @expose("/results/<key>/")
+    @log_this
+    def results(self, key):
+        """Serves a key off of the results backend"""
+        blob = results_backend.get(key)
+        if blob:
+            json_payload = zlib.decompress(blob)
+            obj = json.loads(json_payload)
+            db_id = obj['query']['dbId']
+            session = db.session()
+            mydb = session.query(models.Database).filter_by(id=db_id).first()
+
+            if not self.database_access(mydb):
+                json_error_response(
+                    get_database_access_error_msg(mydb.database_name))
+
+            return Response(
+                json_payload,
+                status=202,
+                mimetype="application/json")
+        else:
+            return Response(
+                json.dumps({
+                    'error': (
+                        "Data could not be retrived. You may want to "
+                        "re-run the query."
+                    )
+                }),
+                status=202,
```

Comment:
s/202/410 ?
`The requested resource is no longer available at the server and no forwarding address is known. `